### PR TITLE
Add scaling and other enhancements to Streamlit UI for post-processing

### PR DESCRIPTION
**Pull request by pineapple-cat**
Addresses:

- [x] #301
- [x] #268
- [x] Help text popups for some fields.
- [x] Streamlit warnings for invalid actions.
- [x] Debug extra columns.
- [x] Debug custom filter values.

### CODE REVIEW

**Review by ilectra — APPROVED** [review]

**Comment by ilectra** [comment]
I'm not sure I understand all the bugfixes, but it works beautifully!

File: post-processing/config_handler.py [hunk 1]
```diff
@@ -102,11 +102,24 @@ def parse_filters(self):
 
         # filters
         if self.filters:
-            self.and_filters = self.filters["and"] if self.filters.get("and") else []
-            self.or_filters = self.filters["or"] if self.filters.get("or") else []
+            # FIXME: consider a better way of doing this
+            # use hashable tuples to remove duplicate filters
```
[the remark this answers is not in the document]
Comment (pineapple-cat) [reply post-processing/config_handler.py:106]: Sets disrupt element ordering, and we definitely don't want to do that - especially not for series.

File: post-processing/config_handler.py [hunk 2]
```diff
@@ -102,11 +102,24 @@ def parse_filters(self):
...
+            self.and_filters = (list(dict.fromkeys([tuple(f) for f in self.filters["and"]]))
+                                if self.filters.get("and") else [])
+            self.or_filters = (list(dict.fromkeys([tuple(f) for f in self.filters["or"]]))
+                               if self.filters.get("or") else [])
+            # convert back to lists to maintain mutability
+            self.and_filters = [list(f) for f in self.and_filters]
+            self.or_filters = [list(f) for f in self.or_filters]
+            # FIXME: consider the purpose of keeping multiple filter lists
```
[the remark this answers is not in the document]
Comment (pineapple-cat) [reply post-processing/config_handler.py:114]: Originally `and_filters` and `or_filters` were intended to be the primary internal variables used as they had cleaner names, but when working on the UI I found that it was easier to work with `filters["and"]` and `filters["or"]` due to Streamlit's key system.

[the remark this answers is not in the document]
Comment (ilectra) [reply post-processing/config_handler.py:114]: I see.... Well, open another issue if you think it makes sense to keep only `filters[]` , it doesn't have to be resolved in this PR.

File: post-processing/config_handler.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ilectra) [reply post-processing/config_handler.py:106]: Oh, right, we don't. Again, open another issue to consider later, no need to fix now.

**Pull request merged**